app/assets/javascripts/discourse/tests/integration/components/dialog-holder-test.js

```diff
@@ -364,4 +364,28 @@ module("Integration | Component | dialog-holder", function (hooks) {
     assert.notOk(query(".dialog-footer"), "no footer");
     assert.notOk(query(".dialog-header"), "no header");
   });
+
+  test("delete confirm", async function (assert) {
+    await render(hbs`<DialogHolder />`);
+
+    this.dialog.deleteConfirm({ message: "A delete confirm message" });
+    await settled();
```

Comment:
What is this awaiting for?

Reply:
For the dialog to finish rendering. It looks like the dialog service has a pending timer. All the other tests for this service use `await settled()` too.